From dff7c06e3ffb82c5af9918b14c0fae32bbcafc0d Mon Sep 17 00:00:00 2001
From: Anilkumar Kolli <akolli@codeaurora.org>
Date: Wed, 10 Feb 2021 15:26:54 +0530
Subject: [PATCH] ath11k: Fix ce interrupt count

CE interrupt count is based on target_ce_count.
Use target_ce_count to request and process ce interrupts.

Fixes: c5000fe803 ("ath11k: Add ipq6018 support")

Signed-off-by: Anilkumar Kolli <akolli@codeaurora.org>
---
 drivers/net/wireless/ath/ath11k/ahb.c | 12 ++++++------
 drivers/net/wireless/ath/ath11k/pci.c | 14 +++++++-------
 2 files changed, 13 insertions(+), 13 deletions(-)

--- a/drivers/net/wireless/ath/ath11k/ahb.c
+++ b/drivers/net/wireless/ath/ath11k/ahb.c
@@ -176,7 +176,7 @@ static void ath11k_ahb_kill_tasklets(str
 {
 	int i;
 
-	for (i = 0; i < ab->hw_params.ce_count; i++) {
+	for (i = 0; i < ab->hw_params.target_ce_count; i++) {
 		struct ath11k_ce_pipe *ce_pipe = &ab->ce.ce_pipe[i];
 
 		if (ath11k_ce_get_attr_flags(ab, i) & CE_ATTR_DIS_INTR)
@@ -289,7 +289,7 @@ static void ath11k_ahb_sync_ce_irqs(stru
 	int i;
 	int irq_idx;
 
-	for (i = 0; i < ab->hw_params.ce_count; i++) {
+	for (i = 0; i < ab->hw_params.target_ce_count; i++) {
 		if (ath11k_ce_get_attr_flags(ab, i) & CE_ATTR_DIS_INTR)
 			continue;
 
@@ -317,7 +317,7 @@ static void ath11k_ahb_ce_irqs_enable(st
 {
 	int i;
 
-	for (i = 0; i < ab->hw_params.ce_count; i++) {
+	for (i = 0; i < ab->hw_params.target_ce_count; i++) {
 		if (ath11k_ce_get_attr_flags(ab, i) & CE_ATTR_DIS_INTR)
 			continue;
 		ath11k_ahb_ce_irq_enable(ab, i);
@@ -328,7 +328,7 @@ static void ath11k_ahb_ce_irqs_disable(s
 {
 	int i;
 
-	for (i = 0; i < ab->hw_params.ce_count; i++) {
+	for (i = 0; i < ab->hw_params.target_ce_count; i++) {
 		if (ath11k_ce_get_attr_flags(ab, i) & CE_ATTR_DIS_INTR)
 			continue;
 		ath11k_ahb_ce_irq_disable(ab, i);
@@ -442,7 +442,7 @@ static void ath11k_ahb_free_irq(struct a
 	int irq_idx;
 	int i;
 
-	for (i = 0; i < ab->hw_params.ce_count; i++) {
+	for (i = 0; i < ab->hw_params.target_ce_count; i++) {
 		if (ath11k_ce_get_attr_flags(ab, i) & CE_ATTR_DIS_INTR)
 			continue;
 		irq_idx = ATH11K_IRQ_CE0_OFFSET + i;
@@ -674,7 +674,7 @@ static int ath11k_ahb_config_irq(struct
 	int ret = 0;
 
 	/* Configure CE irqs */
-	for (i = 0; i < ab->hw_params.ce_count; i++) {
+	for (i = 0; i < ab->hw_params.target_ce_count; i++) {
 		struct ath11k_ce_pipe *ce_pipe = &ab->ce.ce_pipe[i];
 
 		if (ath11k_ce_get_attr_flags(ab, i) & CE_ATTR_DIS_INTR)
--- a/drivers/net/wireless/ath/ath11k/pci.c
+++ b/drivers/net/wireless/ath/ath11k/pci.c
@@ -495,7 +495,7 @@ void ath11k_pci_get_ce_msi_idx(struct at
 {
 	u32 i, msi_data_idx;
 
-	for (i = 0, msi_data_idx = 0; i < ab->hw_params.ce_count; i++) {
+	for (i = 0, msi_data_idx = 0; i < ab->hw_params.target_ce_count; i++) {
 		if (ath11k_ce_get_attr_flags(ab, i) & CE_ATTR_DIS_INTR)
 			continue;
 
@@ -565,7 +565,7 @@ void ath11k_pci_free_irq(struct ath11k_b
 {
 	int i, irq_idx;
 
-	for (i = 0; i < ab->hw_params.ce_count; i++) {
+	for (i = 0; i < ab->hw_params.target_ce_count; i++) {
 		if (ath11k_ce_get_attr_flags(ab, i) & CE_ATTR_DIS_INTR)
 			continue;
 		irq_idx = ATH11K_PCI_IRQ_CE0_OFFSET + i;
@@ -596,7 +596,7 @@ static void ath11k_pci_ce_irqs_disable(s
 {
 	int i;
 
-	for (i = 0; i < ab->hw_params.ce_count; i++) {
+	for (i = 0; i < ab->hw_params.target_ce_count; i++) {
 		if (ath11k_ce_get_attr_flags(ab, i) & CE_ATTR_DIS_INTR)
 			continue;
 		ath11k_pci_ce_irq_disable(ab, i);
@@ -608,7 +608,7 @@ static void ath11k_pci_sync_ce_irqs(stru
 	int i;
 	int irq_idx;
 
-	for (i = 0; i < ab->hw_params.ce_count; i++) {
+	for (i = 0; i < ab->hw_params.target_ce_count; i++) {
 		if (ath11k_ce_get_attr_flags(ab, i) & CE_ATTR_DIS_INTR)
 			continue;
 
@@ -858,7 +858,7 @@ static int ath11k_config_msi_irq(struct
 		return ret;
 
 	/* Configure CE irqs */
-	for (i = 0, msi_data_idx = 0; i < ab->hw_params.ce_count; i++) {
+	for (i = 0, msi_data_idx = 0; i < ab->hw_params.target_ce_count; i++) {
 		if (ath11k_ce_get_attr_flags(ab, i) & CE_ATTR_DIS_INTR)
 			continue;
 
@@ -924,7 +924,7 @@ static void ath11k_pci_ce_irqs_enable(st
 {
 	int i;
 
-	for (i = 0; i < ab->hw_params.ce_count; i++) {
+	for (i = 0; i < ab->hw_params.target_ce_count; i++) {
 		if (ath11k_ce_get_attr_flags(ab, i) & CE_ATTR_DIS_INTR)
 			continue;
 		ath11k_pci_ce_irq_enable(ab, i);
@@ -1150,7 +1150,7 @@ static void ath11k_pci_kill_tasklets(str
 {
 	int i;
 
-	for (i = 0; i < ab->hw_params.ce_count; i++) {
+	for (i = 0; i < ab->hw_params.target_ce_count; i++) {
 		struct ath11k_ce_pipe *ce_pipe = &ab->ce.ce_pipe[i];
 
 		if (ath11k_ce_get_attr_flags(ab, i) & CE_ATTR_DIS_INTR)
